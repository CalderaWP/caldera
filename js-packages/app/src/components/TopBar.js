import React, { Component, Fragment } from 'react';
import calderaWpLogo from './logos/Logo-CalderaWP-DarkBG.svg';
import calderaWpIcon from './logos/icons/Icon-CalderaWP-DarkBG.svg';

import calderaFormsLogo from './logos/Logo-CalderaForms-DarkBG.svg';
import calderaFormsIcon from './logos/icons/Icon-CalderaForms-DarkBG-Alt.svg';

import calderaPayLogo from './logos/Logo-CalderaPay-DarkBG.svg';
import calderaPayIcon from './logos/icons/Icon-CalderaPay-DarkBG-Alt.svg';

import calderaSocialLogo from './logos/Logo-CalderaSocial-DarkBG.svg';
import calderaSocialIcon from './logos/icons/Icon-Social-DarkBG-Alt.svg';

import PropTypes from 'prop-types';
const merge = require('deepmerge');

export class TopBar extends Component {
	state = {
		activeItem: 'home'
	};

	menuItems = [
		{
			key: 'home',
			title: 'CalderaWP',
			icon: calderaWpIcon,
			logo: calderaWpLogo
		},
		{
			key: 'calderaForms',
			title: 'Caldera Forms',
			icon: calderaFormsIcon,
			logo: calderaFormsLogo
		},
		{
			key: 'calderaPay',
			title: 'Caldera Pay',
			icon: calderaPayIcon,
			logo: calderaPayLogo
		},
		{
			key: 'calderaSocial',
			title: 'Caldera Social',
			icon: calderaSocialIcon,
			logo: calderaSocialLogo
		}
	];

	styles = merge(
		{
			header: {
				height: 54,
				display: 'flex',
				flexDirection: 'row',
				alignItems: 'center',
				padding: 8,
				backgroundColor: '#333333',
				color: '#ffffff'
			},
			heading: {
				fontSize: 20,
				fontWeight: 'normal'
			}
		},
		this.props.styles || {}
	);

	setActive(activeItem) {
		this.props.onChangeActive(activeItem);
		this.setState({ activeItem });
	}

	activeItem() {
		return this.menuItems.find(item => this.state.activeItem === item.key);
	}

	render() {
		const activeItem = this.activeItem();

		return (
			<div style={this.styles.header}>
				<img
					key={'active-item'}
					style={{
						width: '200px',
						height: '75px',
						padding: '15px'
					}}
					src={activeItem.logo}
					alt={`${activeItem.title} Logo`}
				/>

				<nav
					key={'main-nav'}
				>
					{this.menuItems.map(menuItem => {
						const { icon, title, key } = menuItem;
						if (activeItem.key === key) {
							return <Fragment key={key} />;
						}
						return (
							<a
								key={key}
								onClick={() => {
									this.setActive(key);
								}}
							>
								<img
									style={{
										width: '200px',
										height: '75px',
										padding: '15px'
									}}
									src={icon}
									alt={`${title} Icon`}
								/>
							</a>
						);
					})}
				</nav>
			</div>
		);
	}
}

TopBar.propTypes = {
	onChangeActive: PropTypes.func
};
const noop = () => {};

TopBar.defaultProps = {
	onChangeActive: noop
};
