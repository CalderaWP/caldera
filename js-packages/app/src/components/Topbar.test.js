import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';
import {TopBar} from './TopBar';
describe( 'Topbar', () => {
	it( 'Matches snpashot' , ()=> {
		const component = renderer.create(
			<TopBar/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	});

	it( 'Calls route change handler' , ()=> {
		const onChangeActive = jest.fn();
		const component = shallow(
			<TopBar
				onChangeActive={onChangeActive}
			/>
		);
		component.find( 'a' ).first().simulate('click');
		expect( onChangeActive.mock.calls.length ).toBe(1);
	});
});
