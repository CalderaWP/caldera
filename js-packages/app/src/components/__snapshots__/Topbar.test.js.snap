// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topbar Matches snpashot 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#333333",
      "color": "#ffffff",
      "display": "flex",
      "flexDirection": "row",
      "height": 54,
      "padding": 8,
    }
  }
>
  <img
    alt="CalderaWP Logo"
    src="Logo-CalderaWP-DarkBG.svg"
    style={
      Object {
        "height": "75px",
        "padding": "15px",
        "width": "200px",
      }
    }
  />
  <nav>
    <a
      onClick={[Function]}
    >
      <img
        alt="Caldera Forms Icon"
        src="Icon-CalderaForms-DarkBG-Alt.svg"
        style={
          Object {
            "height": "75px",
            "padding": "15px",
            "width": "200px",
          }
        }
      />
    </a>
    <a
      onClick={[Function]}
    >
      <img
        alt="Caldera Pay Icon"
        src="Icon-CalderaPay-DarkBG-Alt.svg"
        style={
          Object {
            "height": "75px",
            "padding": "15px",
            "width": "200px",
          }
        }
      />
    </a>
  </nav>
</div>
`;
